client/src/views/Simulation/components/ModelPanel.vue

```diff
@@ -81,6 +78,31 @@
       { name: capitalize(VIEWS.FUNCTIONAL), id: VIEWS.FUNCTIONAL },
     ];
 
+    onNodeClick (selected: Graph.GraphNodeInterface): void {
+      const selectedName = selected.label;
+      this.getSimParameters.forEach(parameter => {
+        if (parameter.metadata.name === selectedName) {
+          parameter.hidden = !parameter.hidden;
+        }
+      });
+      this.getSimVariables.forEach(variable => {
+        if (variable.metadata.name === selectedName) {
+          variable.hidden = !variable.hidden;
+        }
+      });
+    }
+
+    get highlighted (): Graph.GraphInterface {
+      const highlightedLabels = [
+        ...this.getSimParameters.filter(parameter => !parameter.hidden).map(parameter => parameter.metadata.name),
```

Comment:
Maybe you can check with the `uid` property instead of `metadata.name`?
A bit like I did here https://github.com/uncharted-aske/HMI/blob/main/client/src/store/modules/simulation.ts#L165

Reply:
unfortunately this won't work. i double checked and the data being rendered by d3 (the data available here) does not contain a `uid` property. heres a sample:

<img width="441" alt="Screen Shot 2021-07-19 at 1 47 27 PM" src="https://user-images.githubusercontent.com/14062458/126203956-649bad7c-1b82-4af0-b226-bedce2569c16.png">